src/oaks_speech_obj.c

```diff
@@ -0,0 +1,187 @@
+
+#include "oaks_speech_internal.h"
+#include "vram_transfer_manager.h"
+
+static const UnkStruct_0200D2B4 sSpriteTemplates[] = {
+    {
+     .resourceSet = 0,
+     .x           = 128,
+     .y           = 50,
+     .z           = 0,
+     .animSeqNo   = 0,
+     .rotation    = 2,
+     .unk_10      = 1,
+     .whichScreen = NNS_G2D_VRAM_TYPE_2DMAIN,
+     .unk_18      = 0,
+     .unk_1C      = 0,
+     .unk_20      = 0,
+     .unk_24      = 0,
+     },
+    {
+     .resourceSet = 1,
+     .x           = 64,
+     .y           = 104,
+     .z           = 0,
+     .animSeqNo   = 0,
+     .rotation    = 0,
+     .unk_10      = 0,
+     .whichScreen = NNS_G2D_VRAM_TYPE_2DSUB,
+     .unk_18      = 0,
+     .unk_1C      = 0,
+     .unk_20      = 0,
+     .unk_24      = 0,
+     },
+    {
+     .resourceSet = 2,
+     .x           = 192,
+     .y           = 104,
+     .z           = 0,
+     .animSeqNo   = 0,
+     .rotation    = 1,
+     .unk_10      = 1,
+     .whichScreen = NNS_G2D_VRAM_TYPE_2DSUB,
+     .unk_18      = 0,
+     .unk_1C      = 0,
+     .unk_20      = 0,
+     .unk_24      = 0,
+     },
+    {
+     .resourceSet = 3,
+     .x           = 256,
+     .y           = 192,
+     .z           = 0,
+     .animSeqNo   = 0,
+     .rotation    = 0,
+     .unk_10      = 3,
+     .whichScreen = NNS_G2D_VRAM_TYPE_2DSUB,
+     .unk_18      = 0,
+     .unk_1C      = 0,
+     .unk_20      = 0,
+     .unk_24      = 0,
+     },
+    {
+     .resourceSet = 4,
+     .x           = 0,
+     .y           = 0,
+     .z           = 0,
+     .animSeqNo   = 1,
+     .rotation    = 0,
+     .unk_10      = 4,
+     .whichScreen = NNS_G2D_VRAM_TYPE_2DSUB,
+     .unk_18      = 0,
+     .unk_1C      = 0,
+     .unk_20      = 0,
+     .unk_24      = 0,
+     },
+    {
+     .resourceSet = 5,
+     .x           = 160,
+     .y           = 80,
+     .z           = 0,
+     .animSeqNo   = 0,
+     .rotation    = 0,
+     .unk_10      = 4,
+     .whichScreen = NNS_G2D_VRAM_TYPE_2DMAIN,
+     .unk_18      = 0,
+     .unk_1C      = 0,
+     .unk_20      = 0,
+     .unk_24      = 0,
+     },
+};
+
+void OakSpeech_InitSpriteEngine(OakSpeechData *data) {
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_ON);
+    GfGfx_EngineBTogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_ON);
+    GF_CreateVramTransferManager(0x20, HEAP_ID_OAKS_SPEECH);
+    data->spriteRenderer   = SpriteRenderer_Create(HEAP_ID_OAKS_SPEECH);
+    data->spriteGfxHandler = SpriteRenderer_CreateGfxHandler(data->spriteRenderer);
+
+    {
+        // extern const OamManagerParam ov53_021E8764;
+        OamManagerParam oamManagerParam = {
+            .fromOBJmain    = 0,
+            .numOBJmain     = 128,
+            .fromAffineMain = 0,
+            .numAffineMain  = 32,
+            .fromOBJsub     = 0,
+            .numOBJsub      = 128,
+            .fromAffineSub  = 0,
+            .numAffineSub   = 32,
+        };
+
+        // extern const OamCharTransferParam ov53_021E8750;
```

Comment:
can remove comment

Reply:
the data inlining is in part fakematched so the comment can stay